infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -210,33 +211,28 @@ private void createSessions() {
         Set<LogReplicationSession> outgoingSessionsToAdd = new HashSet<>();
 
         try {
+            String localClusterId = topology.getLocalClusterDescriptor().getClusterId();
+
             IRetry.build(IntervalRetry.class, () -> {
                 try (TxnContext txn = corfuStore.txn(LogReplicationMetadataManager.NAMESPACE)) {
 
-                    for (ClusterDescriptor sourceCluster : topology.getSourceClusters().values()) {
-                        for (ClusterDescriptor sinkCluster : topology.getSinkClusters().values()) {
-
-                            // TODO(V2): for now only creating sessions for FULL TABLE replication model (assumed as default)
-                            LogReplicationSession session = LogReplicationSession.newBuilder()
-                                .setSourceClusterId(sourceCluster.clusterId)
-                                .setSinkClusterId(sinkCluster.clusterId)
-                                .setSubscriber(getDefaultSubscriber())
-                                .build();
-
-                            if (!sessions.contains(session)) {
-                                if (session.getSourceClusterId() == topology.getLocalClusterDescriptor().getClusterId()) {
-                                    sessionsToAdd.add(session);
-                                    outgoingSessionsToAdd.add(session);
-                                    metadataManager.addSession(txn, session, topology.getTopologyConfigId(), false);
-                                } else if (session.getSinkClusterId() == topology.getLocalClusterDescriptor().getClusterId()) {
-                                    sessionsToAdd.add(session);
-                                    incomingSessionsToAdd.add(session);
-                                    metadataManager.addSession(txn, session, topology.getTopologyConfigId(), true);
-                                } else {
-                                    log.info("Session {} does not contain current node {} - skipping", session,
-                                        topology.getLocalClusterDescriptor().getClusterId());
-                                }
-                            }
+                    for(ClusterDescriptor remoteSinkCluster : topology.getRemoteSinkClusters().values()) {
```

Comment:
This logic seems different from the nested loop for source and sink. Can you please add a comment if this is intentional?

Reply:
sync'ed up offline.